File: Algorithms/1381_Design_a_Stack_With_Increment_Operation/CustomStack.scala
class CustomStack(_maxSize: Int) {

  private var size = 0
  private val stk = Array.fill(_maxSize)(0)

  def push(x: Int): Unit = if (size < _maxSize) {
    stk(size) = x
    size += 1
  }

  def pop(): Int = if (size == 0) -1 else {
    size -= 1
    stk(size)
  }

  def increment(k: Int, `val`: Int): Unit = Range(0, math.min(k, size)).foreach(stk(_) += `val`)

}

File: Algorithms/1381_Design_a_Stack_With_Increment_Operation/CustomStack.kt
class CustomStack(private val maxSize: Int) {
    private var size = 0
    private val stk = IntArray(maxSize)

    fun push(x: Int) {
        if (size < maxSize) {
            stk[size++] = x
        }
    }

    fun pop(): Int = if (size > 0) stk[--size] else -1

    fun increment(k: Int, `val`: Int) = 0.until(Integer.min(size, k)).forEach { stk[it] += `val` }
}